diff -ruN dietlibc-0.33/include/fcntl.h dietlibc-0.33-changes//include/fcntl.h
--- dietlibc-0.33/include/fcntl.h	2012-11-02 07:17:28.000000000 -0400
+++ dietlibc-0.33-changes//include/fcntl.h	2014-01-09 16:57:06.000000000 -0500
@@ -34,6 +34,7 @@
 #define O_DIRECTORY	0200000	/* must be a directory */
 #define O_NOFOLLOW	0400000 /* don't follow links */
 #define O_NOATIME	01000000
+#define O_CLOEXEC	02000000
 
 #define F_DUPFD		0	/* dup */
 #define F_GETFD		1	/* get close_on_exec */
@@ -654,6 +655,7 @@
 #define AT_SYMLINK_NOFOLLOW	0x100   /* Do not follow symbolic links.  */
 #define AT_REMOVEDIR		0x200   /* Remove directory instead of unlinking file.  */
 #define AT_SYMLINK_FOLLOW	0x400   /* Follow symbolic links.  */
+#define AT_EACCESS		0x200
 
 int openat(int dirfd, const char *pathname, int flags, ...) __THROW;
 int futimesat(int dirfd, const char *pathname, const struct timeval times[2]) __THROW;
diff -ruN dietlibc-0.33/lib/wcstok.c dietlibc-0.33-changes//lib/wcstok.c
--- dietlibc-0.33/lib/wcstok.c	1969-12-31 19:00:00.000000000 -0500
+++ dietlibc-0.33-changes//lib/wcstok.c	2014-01-09 16:37:45.000000000 -0500
@@ -0,0 +1,37 @@
+#include <wchar.h>
+
+wchar_t *wcstok(wchar_t *__restrict__ s, const wchar_t *__restrict__ delim,
+	wchar_t **__restrict__ last)
+{
+	const wchar_t * spanp;
+	wchar_t *tok, c, sc;
+
+	if (s==NULL && (s=*last)==NULL) return NULL;
+
+cont:
+	c = *s++;
+	for (spanp = delim; (sc = *spanp++) != L'\0';) {
+		if (c == sc) goto cont;
+	}
+
+	if (c == L'\0') {
+		*last = NULL;
+		return NULL;
+	}
+	tok = s - 1;
+
+	for (;;) {
+		c = *s++;
+		spanp = delim;
+		do {
+			if ((sc = *spanp++)==c) {
+				if (c == L'\0')
+					s = NULL;
+				else
+					s[-1] = L'\0';
+				*last = s;
+				return tok;
+			}
+		} while (sc != L'\0');
+	}
+}
